taskchampion/taskchampion/src/server/cloud/gcp.rs

```diff
@@ -0,0 +1,392 @@
+        _ => false,
+    }
+}
+
+impl GcpService {
+    pub(in crate::server) fn new(bucket: String) -> Result<Self> {
+        let rt = Runtime::new()?;
+        let config = rt.block_on(ClientConfig::default().with_auth())?;
+        Ok(Self {
+            client: Client::new(config),
+            rt,
+            bucket,
+        })
+    }
+}
+
+impl Service for GcpService {
+    fn put(&mut self, name: &[u8], value: &[u8]) -> Result<()> {
+        let name = String::from_utf8(name.to_vec()).expect("non-UTF8 object name");
+        let upload_type = objects::upload::UploadType::Simple(objects::upload::Media::new(name));
+        self.rt.block_on(self.client.upload_object(
+            &objects::upload::UploadObjectRequest {
+                bucket: self.bucket.clone(),
+                ..Default::default()
+            },
+            value.to_vec(),
+            &upload_type,
+        ))?;
+        Ok(())
+    }
+
+    fn get(&mut self, name: &[u8]) -> Result<Option<Vec<u8>>> {
+        let name = String::from_utf8(name.to_vec()).expect("non-UTF8 object name");
+        let download_res = self.rt.block_on(self.client.download_object(
+            &objects::get::GetObjectRequest {
+                bucket: self.bucket.clone(),
+                object: name,
+                ..Default::default()
+            },
+            &objects::download::Range::default(),
+        ));
+        if is_http_error(404, &download_res) {
+            Ok(None)
+        } else {
+            Ok(Some(download_res?))
+        }
+    }
+
+    fn del(&mut self, name: &[u8]) -> Result<()> {
+        let name = String::from_utf8(name.to_vec()).expect("non-UTF8 object name");
+        let del_res = self.rt.block_on(self.client.delete_object(
+            &objects::delete::DeleteObjectRequest {
+                bucket: self.bucket.clone(),
+                object: name,
+                ..Default::default()
+            },
+        ));
+        if !is_http_error(404, &del_res) {
+            del_res?;
+        }
+        Ok(())
+    }
+
+    fn list<'a>(&'a mut self, prefix: &[u8]) -> Box<dyn Iterator<Item = Result<ObjectInfo>> + 'a> {
+        let prefix = String::from_utf8(prefix.to_vec()).expect("non-UTF8 object prefix");
+        Box::new(ObjectIterator {
+            service: self,
+            prefix,
+            last_response: None,
+            next_index: 0,
+        })
+    }
+
+    fn compare_and_swap(
+        &mut self,
+        name: &[u8],
+        existing_value: Option<Vec<u8>>,
+        new_value: Vec<u8>,
+    ) -> Result<bool> {
+        let name = String::from_utf8(name.to_vec()).expect("non-UTF8 object name");
+        let get_res = self
+            .rt
+            .block_on(self.client.get_object(&objects::get::GetObjectRequest {
+                bucket: self.bucket.clone(),
+                object: name.clone(),
+                ..Default::default()
+            }));
+        let generation = if is_http_error(404, &get_res) {
```

Comment:
If this is a common term of art it may be unnecessary but it sure would have helped me grok this function faster.

```suggestion
        // "generation" is the object's version number
        let generation = if is_http_error(404, &get_res) {
```

Reply:
It's part of the GCS API. I'll add a comment.

Reply:
BTW, I still don't have enough privilege to resolve my own review comments. :roll_eyes: 

To be fair to github though, I can think of an attack that would take advantage of unprivileged resolving of one's own review comments:

1. Malicious actor opens pull request with obvious vulnerability.
2. Malicious actor uses sock puppet account to review PR and comments on vulnerability.
3. Trusted reviewer reviews PR and doesn't comment on vulnerability since it's already been commented on.
4. Sock puppet resolves comment.
5. Malicious actor addresses trusted reviewer's comments.
6. Trusted reviewer approves pull request.
7. Somebody merges it.